pytorch_lightning/trainer/trainer.py

```diff
@@ -320,6 +320,10 @@ def __init__(
         self.predict_loop = PredictLoop(self)
 
         # training state
+        if weights_summary is not None and weights_summary not in ModelSummary.MODES:
+            raise MisconfigurationException(
+                f"`weights_summary` can be None, {', '.join(ModelSummary.MODES)}, but got {weights_summary}"
+            )
```

Comment:
i understand we want to fail early but i dislike that the validation here is repeated again in the implementation of `LightningModule.summarize()`

Reply:
>  i dislike that the validation here is repeated again in the implementation of LightningModule.summarize()

It is there because this method is standalone, it has no relation to the trainer and can be used directly on the `LightningModule` outside of a training script

Reply:
@awaelchli in this case shall I move this validation check for `weights_summary` to the `ModelSummary` constructor? https://github.com/PyTorchLightning/pytorch-lightning/blob/634d83134fea4bb701c24abd5a4a38adb0eddbcd/pytorch_lightning/core/memory.py#L185-L192 ? 

There is one test that asserts that the an invalid `weights_summary` will raise an exception on the Trainer init, which is why I moved this here. 

I can split this out to a separate PR if you'd like

Reply:
What happens to the message in the misconfiguration error?
If using the Trainer, the message should inform about weights_summary, and when using the summary object directly it should inform about the mode.

Reply:
ah good catch! it makes sense to have both checks for debuggability